### PR TITLE
Begin supporting IBM XL builds on Darwin

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

+ It is difficult to debug on ATS-2 machines.  Building and testing on Darwin power9 nodes with the IBM XL compiler and Spectrum MPI provides a similar environment.

### Purpose of Pull Request

* [Fixes Redmine Issue #1948](https://rtt.lanl.gov/redmine/issues/1948)

### Description of changes

+ Provide minor build system updates to support Draco builds for XL on Darwin.
  + The compiler option `--gcc-toolchain` appears to be ATS-2 specific so restrict it to ATS-2 machines.
  + Help the build system identify spectrum-mpi as installed on Darwin.
+ Update the draco developer environment so that X11 is automatically enabled for the salloc alias.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: config/setupMPI.cmake [hunk 1]
```diff
@@ -46,7 +46,8 @@ function( setMPIflavorVer )
   elseif( "${MPIEXEC_EXECUTABLE}" MATCHES "mvapich2")
     set( MPI_FLAVOR "mvapich2" )
   elseif( "${MPIEXEC_EXECUTABLE}" MATCHES "spectrum-mpi" OR
-      "${MPIEXEC_EXECUTABLE}" MATCHES "jsrun" )
+      "${MPIEXEC_EXECUTABLE}" MATCHES "jsrun" OR
+      "${MPIEXEC_EXECUTABLE}" MATCHES "smpi"  )
```
Comment (KineticTheory) [inline config/setupMPI.cmake:50]: On darwin Spectrum-MPI is in a directory that matches "smpi".

File: config/unix-xl.cmake [hunk 2]
```diff
@@ -22,14 +22,57 @@ query_openmp_availability()
...
+    if( EXISTS ${config_file} )
+      string( APPEND CMAKE_C_FLAGS " -F${config_file}" )
+    else()
+      message( FATAL_ERROR
+        "IBM XLC selected (Darwin), but requested config file was not found."
+        "\nconfig_file = ${config_file}" )
+    endif()
+    unset( gcc_version )
+    unset( xl_version )
+    unset( config_file )
```
Comment (KineticTheory) [inline config/unix-xl.cmake:69]: This is a big mess.  XL needs special configure options to actually support C++14.  Evidently, these options are not universal across platforms.  On rzansel, we needed the `--gcc-toolchain=` option, but that isn't supported on Darwin.  On darwin we must use the `-F config` option.  The path and name of the configuration file depends on the version of XL, gcc, cuda and the OS.  Very ugly.

File: config/unix-xlf.cmake [hunk 3]
```diff
@@ -11,18 +11,24 @@
...
+   endif()
+
+   set( CMAKE_Fortran_FLAGS_DEBUG "-O0" )
    set( CMAKE_Fortran_FLAGS_RELWITHDEBINFO
-     "-O3 -qhot=novector -qsimd=auto -qstrict=nans:operationprecision" )
+     "-O3 -qstrict=nans:operationprecision" )
    set( CMAKE_Fortran_FLAGS_RELEASE
      "${CMAKE_Fortran_FLAGS_RELWITHDEBINFO}" )
-   set( CMAKE_Fortran_FLAGS_MINSIZEREL     "${CMAKE_Fortran_FLAGS_RELEASE}" )
+   set( CMAKE_Fortran_FLAGS_MINSIZEREL "${CMAKE_Fortran_FLAGS_RELEASE}" )
```
Comment (KineticTheory) [inline config/unix-xlf.cmake:31]: Remove the `auto` options.  These are causing issues for us.

File: config/unix-xl.cmake (hunk 2 above)
Comment (alexrlongne) [reply config/unix-xl.cmake:69]: Thanks for the explanation, that is awful!

**Comment by alexrlongne** [comment]
LGTM, thank you for the clarifying comments.

**Pull request merged**